File: JavaModule/src/main/java/com/example/lee/secondRound/Searcha2DMatrix.java
package com.example.lee.secondRound;

/**
 * Created by benbendaisy on 5/11/15.
 */
public class Searcha2DMatrix {
    public boolean searchMatrix(int[][] matrix, int target) {
        if (null == matrix) return false;
        int row = matrix.length;
        int col = matrix[0].length;
        int rdx = 0, cdx = col - 1;
        while (rdx < row && matrix[rdx][cdx] <= target) {
            if (matrix[rdx][cdx] == target) return true;
            rdx++;
        }
        if (rdx < row) {
            while (cdx >= 0 && matrix[rdx][cdx] >= target) {
                if (matrix[rdx][cdx] == target) return true;
                cdx--;
            }
        }
        return false;
    }
}

File: python_module/examples/search_a_2d_matrix.py
from typing import List


class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        if not matrix or not matrix[0]:
            return False

        row, col = 0, len(matrix[0]) - 1
        while row < len(matrix) and matrix[row][col] < target:
            if matrix[row][col] == target:
                return True
            row += 1

        while row < len(matrix) and col >= 0 and matrix[row][col] >= target:
            if matrix[row][col] == target:
                return True
            col -= 1

        return False

if __name__ == "__main__":
    matrix = [[1]]
    solution = Solution()
    ret = solution.searchMatrix(matrix, 3)
    print(ret)